app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/Dialogs/ImageFilterDialog.java

```diff
@@ -0,0 +1,155 @@
+        btn_save_filter = (Button) view.findViewById(R.id.btn_save_filter);
+        sp_imgsize.setFocusable(true);
+
+        if (getArguments().getSerializable("imageFilter") != null) {
+            imageFilter = (ImageFilter) getArguments().getSerializable("imageFilter");
+        }
+        //hideSoftKeyboard(view);
+        setupSpAdapter(view);
+
+        getDialog().getWindow().setSoftInputMode(
+                WindowManager.LayoutParams.SOFT_INPUT_STATE_VISIBLE);
+
+        btn_save_filter.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                hideSoftKeyboard(v);
+                finishFilter(v);
+            }
+        });
+
+        return view;
+    }
+
+    public void finishFilter(View view){
+
+        String site = et_img_site.getText().toString();
+        String size = sp_imgsize.getSelectedItem().toString();
+        String color = sp_imgcolor.getSelectedItem().toString();
+        String type = sp_img_type.getSelectedItem().toString();
+
+        imageFilter.setSize(size);
+        imageFilter.setType(type);
+        imageFilter.setColor(color);
+        imageFilter.setSite(site);
+
+        listener.onFinishDialog(imageFilter);
+        dismiss();
+    }
+
+    public void setupSpAdapter(View view){
+
+        ArrayAdapter<CharSequence> adaptercolor = ArrayAdapter.createFromResource(view.getContext(),
+                R.array.sp_img_color, android.R.layout.simple_spinner_item);
+        sp_imgcolor.setAdapter(adaptercolor);
+
+        ArrayAdapter<CharSequence> adaptersize = ArrayAdapter.createFromResource(view.getContext(),
+                R.array.sp_img_sz, android.R.layout.simple_spinner_item);
+        sp_imgsize.setAdapter(adaptersize);
+
+        ArrayAdapter<CharSequence> adaptertype = ArrayAdapter.createFromResource(view.getContext(),
+                R.array.sp_img_type, android.R.layout.simple_spinner_item);
+        sp_img_type.setAdapter(adaptertype);
+
+        if (imageFilter != null){
+            for (int i = 0; i < adaptercolor.getCount(); i++) {
+                if (imageFilter.getColor().equals(adaptercolor.getItem(i).toString())) {
+                    sp_imgcolor.setSelection(i);
+                    break;
+                }
+            }
+            for (int i = 0; i < adaptersize.getCount(); i++) {
+                if (imageFilter.getSize().equals(adaptersize.getItem(i).toString())) {
+                    sp_imgsize.setSelection(i);
+                    break;
+                }
+            }
+            for (int i = 0; i < adaptertype.getCount(); i++) {
+                if (imageFilter.getType().equals(adaptertype.getItem(i).toString())) {
+                    sp_img_type.setSelection(i);
+                    break;
+                }
+            }
+            et_img_site.setText(imageFilter.getSite());
+        }
+    }
+
+    public static void hideSoftKeyboard(View view) {
```

Comment:
@edwinperaza you may want to have this on a separate Helper utility class as this may be used often in other places.